etcdmain/etcd.go

```diff
@@ -379,3 +387,11 @@ func genClusterString(name string, urls types.URLs) string {
 	}
 	return strings.Join(addrs, ",")
 }
+
+// handleInterrupt calls the handler on receiving a SIGTERM.
+func handleInterrupt(h func()) {
```

Comment:
can we print out the received signal and print out the actions?

so etcd will not silently shutdown itself from the view of the log.